File: Backjoon2231/Backjoon2231.java
package Backjoon2231;

import java.util.Scanner;
public class Backjoon2231{
    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        for(int i = 1; i < N; i++){
            int temp = i;
            int num = i;
            while(temp > 0){
                num += temp % 10;
                temp /= 10;
            }
            if(num == N){
                System.out.print(i);
                N = 0;
                break;
            }
        }
        sc.close();
        if(N != 0)
            System.out.print("0");
    }
}

File: Backjoon2231/Backjoon2231.py
N = 216
for i in range(1, N):
    temp = i
    num = i
    while(temp > 0):
        num += temp % 10
        temp //= 10
    if(num == N):
        print(i)
        N = 0
        break
if(N != 0):
    print("0")